modules/database/database.bal

```diff
@@ -588,22 +603,29 @@ public isolated transactional function getDataTypesSummary(int experimentId) ret
     return dataSummary;
 }
 
-public isolated transactional function getDataList(int experimentId, boolean all = true, int 'limit = 100, int offset = 0) returns ExperimentDataFull[]|error {
-    var baseQuery = `SELECT dataId, experimentId, name, version, location, type, contentType 
+public isolated transactional function getDataList(int experimentId, string? search, boolean all = true, int 'limit = 100, int offset = 0, int? 'ascending = 1) returns ExperimentDataFull[]|error {
+    log:printDebug(">>>>>" + 'ascending.toString());
+    sql:ParameterizedQuery baseQuery = `SELECT dataId, experimentId, name, version, location, type, contentType 
                      FROM ExperimentData WHERE experimentId=${experimentId} `;
-    var baseQuerySuffix = `ORDER BY name ASC, version DESC 
-                           LIMIT ${'limit} OFFSET ${offset};`;
-
-    stream<ExperimentDataFull, sql:Error?> experimentData;
-    if all {
-        experimentData = experimentDB->query(check new ConcatQuery(baseQuery, baseQuerySuffix));
-    } else {
-        var extraFilter = `AND version=(SELECT MAX(t2.version)
+    if search != () && search != "" {
+        string searchString = "%" + search + "%";
```

Comment:
Possibly refactor this + the if above into a helper function that returns the LIKE criteria or nil to normalize () and "".